Add GenericPackagesService

sirlatrom opened the pull request:
Fixes #1100.

TODO:
  - [x] Tests
  - [ ] Docs

neomantra commented:
@sirlatrom  Thanks for this.  I have continued work on your PR here:
https://github.com/neomantra/go-gitlab/tree/nm-1100

I rebased your PR and added a commit which completes it.
I didn't do a new PR as it is mostly your work.  If you want me to, just let me know.

sirlatrom commented:
> @sirlatrom  Thanks for this.  I have continued work on your PR here:
> https://github.com/neomantra/go-gitlab/tree/nm-1100
> 
> I rebased your PR and added a commit which completes it.
> I didn't do a new PR as it is mostly your work.  If you want me to, just let me know.

Thanks for carrying this! Feel free to create a PR 🤝

svanharmelen commented:
I'll close this in one favor of #1180
